validations/src/validations.rs

```diff
@@ -244,23 +247,50 @@ pub fn validate_rad_request(rad_request: &RADRequest) -> Result<(), failure::Err
     if retrieval_paths.is_empty() {
         return Err(DataRequestError::NoRetrievalSources.into());
     }
+    let mut is_rng = false;
     for path in retrieval_paths {
+        if path.kind == RADType::Rng {
+            if path.url.is_empty() || path.script.is_empty() {
+                is_rng = true;
+                break;
+            } else {
+                return Err(DataRequestError::InvalidRngRequest.into());
+            }
+        }
+
         // If the sources are empty the data request is set as invalid
         if path.url.is_empty() {
             return Err(DataRequestError::NoRetrievalSources.into());
         }
```

Comment:
Won't this cause an early return for any retrieval path / data source of RADType::Rng that has both `url` and `script` being empty?

Reply:
In the rng case it does never reach this line because there is a `break` in the previous condition. But that limits rng requests to one retrieval source, so we need to change this logic.

Reply:
If it is a RADType::Rng, it will always return earlier in the previous branchs (break or DataRequestError::InvalidRngRequest)